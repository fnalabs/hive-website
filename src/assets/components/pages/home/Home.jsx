import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ReactGA from 'react-ga'

import Consent from 'contexts/Consent'

import { Header } from 'layout'

import { TopHero } from './TopHero'
import { MidHero } from './MidHero'
import NextPageHero from '../NextPageHero'

import meta from 'metadata'

export default class Home extends Component {
  static contextType = Consent
  static contextTypes = {
    isConsent: PropTypes.bool
  }

  componentDidMount () {
    if (this.context.isConsent) {
      ReactGA.pageview(this.props.location.pathname, undefined, meta.common.siteName)
    }
  }

  render () {
    const { description, url } = meta['/']

    return (
      <>
        <Header title={meta.common.siteName} description={description} url={url} />

        <TopHero />
        <MidHero />

        <NextPageHero toRight='/overview' />
      </>
    )
  }
}
