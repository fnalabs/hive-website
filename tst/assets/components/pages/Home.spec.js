/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import { MemoryRouter } from 'react-router'
import { Cookies } from 'react-cookie-consent'
import ReactGA from 'react-ga'

import Home from '../../../../src/assets/components/pages/home/Home.jsx'

jest.mock('react-ga')

describe('<Home />', () => {
  test('should render homepage static content', () => {
    Cookies.get = jest.fn().mockReturnValue(false)

    const tree = renderer.create(<MemoryRouter><Home location={{ pathname: '/' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  test('should render homepage static content and report page view', () => {
    Cookies.get = jest.fn().mockReturnValue(true)

    const tree = renderer.create(<MemoryRouter><Home location={{ pathname: '/' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
