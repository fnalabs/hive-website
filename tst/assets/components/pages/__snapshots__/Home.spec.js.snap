// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render homepage static content 1`] = `
Array [
  <section
    className="hero is-fullheight is-dark is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <div
          className="tile is-ancestor"
        >
          <div
            className="tile is-parent"
          >
            <div
              className="tile is-child is-6 is-vertical has-text-centered"
            >
              <h1
                className="title"
              >
                <svg
                  className="svg-inline"
                  height="32"
                  role="img"
                  viewBox="0 0 117.31758 135.46667"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m 58.658792,6.7627869 52.802038,30.4852731 0,60.970543 L 58.658789,128.70387 5.8567507,98.2186 l 1.7e-6,-60.970543 z"
                    fill="none"
                    stroke="#daa520"
                    strokeWidth="12"
                  />
                </svg>
                Hive
                <sup>
                  io
                </sup>
              </h1>
              <h2
                className="subtitle"
              >
                A reactive, cloud-native framework for building microservices
              </h2>
              <h2
                className="subtitle"
              >
                <svg
                  className="svg-link"
                  height="32"
                  role="img"
                  viewBox="-12.7 -12.7 1024.4 1024.4"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M528.7 545.9c-42 42-42 110.1.0 152.1s110.1 42 152.1.0 42-110.1.0-152.1-110.1-42-152.1.0zm113.7 113.8c-20.8 20.8-54.5 20.8-75.3.0s-20.8-54.5.0-75.3 54.5-20.8 75.3.0c20.8 20.7 20.8 54.5.0 75.3zm36.6-643-65.9 65.9c-12.9 12.9-12.9 34.1.0 47l257.3 257.3c12.9 12.9 34.1 12.9 47 0l65.9-65.9c12.9-12.9 12.9-34.1.0-47L725.9 16.7c-12.9-12.9-34-12.9-46.9.0zM217.3 858.8c11.7-11.7 11.7-30.8.0-42.5l-33.5-33.5c-11.7-11.7-30.8-11.7-42.5.0L72.1 852l-.1.1-19-19c-10.5-10.5-27.6-10.5-38 0-10.5 10.5-10.5 27.6.0 38l114 114c10.5 10.5 27.6 10.5 38 0s10.5-27.6.0-38l-19-19 .1-.1 69.2-69.2z"
                    fill="#f5a800"
                  />
                  <path
                    d="M565.9 205.9 419.5 352.3c-13 13-13 34.4.0 47.4l90.4 90.4c63.9-46 153.5-40.3 211 17.2l73.2-73.2c13-13 13-34.4.0-47.4L613.3 205.9c-13-13.1-34.4-13.1-47.4.0zm-94 322.3-53.4-53.4c-12.5-12.5-33-12.5-45.5.0L184.7 663.2c-12.5 12.5-12.5 33 0 45.5l106.7 106.7c12.5 12.5 33 12.5 45.5.0L458 694.1c-25.6-52.9-21-116.8 13.9-165.9z"
                    fill="#425cc7"
                  />
                </svg>
                Now integrated with 
                <a
                  aria-current={null}
                  href="/telemetry"
                  onClick={[Function]}
                >
                  OpenTelemetry
                </a>
              </h2>
            </div>
            <div
              className="tile is-child is-6"
            >
              <div
                className="svg-animation-container"
              >
                <div
                  className="svg-animation is-mongo-svg is-svg-1"
                />
                <div
                  className="svg-animation is-hive-svg is-svg-2"
                />
                <div
                  className="svg-animation is-envoy-svg is-svg-3"
                />
                <div
                  className="svg-animation is-elastic-svg is-svg-4"
                />
                <div
                  className="svg-animation is-hive-svg is-svg-5"
                />
                <div
                  className="svg-animation is-hive-svg is-svg-6"
                />
                <div
                  className="svg-animation is-envoy-svg is-svg-7"
                />
                <div
                  className="svg-animation is-redis-svg is-svg-8"
                />
                <div
                  className="svg-animation is-kafka-svg is-svg-9"
                />
                <div
                  className="svg-animation is-hive-svg is-svg-10"
                />
                <div
                  className="svg-animation is-hive-svg is-svg-11"
                />
                <div
                  className="svg-animation is-hive-svg is-svg-12"
                />
                <div
                  className="svg-animation is-envoy-svg is-svg-13"
                />
                <div
                  className="svg-animation is-envoy-svg is-svg-14"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="hero-foot has-text-centered"
    >
      <span
        className="icon is-medium"
      >
        <svg
          fill="#fff"
          height="32"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 256.3L7 120.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0L313 86.3c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.4 9.5-24.6 9.5-34 .1zm34 192l136-136c9.4-9.4 9.4-24.6 0-33.9l-22.6-22.6c-9.4-9.4-24.6-9.4-33.9 0L160 352.1l-96.4-96.4c-9.4-9.4-24.6-9.4-33.9 0L7 278.3c-9.4 9.4-9.4 24.6 0 33.9l136 136c9.4 9.5 24.6 9.5 34 .1z"
          />
        </svg>
      </span>
    </div>
  </section>,
  <section
    className="hero is-medium is-dark neon-tiles"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <div
          className="tile is-ancestor"
        >
          <div
            className="tile is-8"
          >
            <div
              className="tile is-vertical"
            >
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child"
                >
                  <article
                    className="card m-6 is-reactive"
                  >
                    <div
                      className="card-image"
                    >
                      <figure
                        className="image is-96x96 has-text-centered"
                      >
                        <svg
                          height="96"
                          role="img"
                          viewBox="0 0 320 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M295.973 160H180.572L215.19 30.184C219.25 14.956 207.756 0 192 0H56C43.971 0 33.8 8.905 32.211 20.828l-31.996 240C-1.704 275.217 9.504 288 24.004 288h118.701L96.646 482.466C93.05 497.649 104.659 512 119.992 512c8.35 0 16.376-4.374 20.778-11.978l175.973-303.997c9.244-15.967-2.288-36.025-20.77-36.025z"
                          />
                        </svg>
                      </figure>
                    </div>
                    <div
                      className="card-content content"
                    >
                      <h3
                        className="has-text-centered"
                      >
                        Responsive
                      </h3>
                      <p>
                        Hive
                        <sup>
                          io
                        </sup>
                         is able to give lightning fast response times by implementing and supporting microservices with Node.js and Fastify.
                      </p>
                    </div>
                  </article>
                </div>
              </div>
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child"
                >
                  <article
                    className="card m-6 is-reactive"
                  >
                    <div
                      className="card-image"
                    >
                      <figure
                        className="image is-96x96 has-text-centered"
                      >
                        <svg
                          height="96"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 180V56c0-13.3 10.7-24 24-24h124c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H64v84c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12zM288 44v40c0 6.6 5.4 12 12 12h84v84c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12V56c0-13.3-10.7-24-24-24H300c-6.6 0-12 5.4-12 12zm148 276h-40c-6.6 0-12 5.4-12 12v84h-84c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h124c13.3 0 24-10.7 24-24V332c0-6.6-5.4-12-12-12zM160 468v-40c0-6.6-5.4-12-12-12H64v-84c0-6.6-5.4-12-12-12H12c-6.6 0-12 5.4-12 12v124c0 13.3 10.7 24 24 24h124c6.6 0 12-5.4 12-12z"
                          />
                        </svg>
                      </figure>
                    </div>
                    <div
                      className="card-content content"
                    >
                      <h3
                        className="has-text-centered"
                      >
                        Resilient
                      </h3>
                      <p>
                        Each microservice contains and isolates aggregates in your domain model to ensure that you are still capable of serving traffic, even during a catastrophic event or unexpected high volume.
                      </p>
                    </div>
                  </article>
                </div>
              </div>
            </div>
            <div
              className="tile is-vertical"
            >
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child"
                >
                  <article
                    className="card m-6 is-reactive"
                  >
                    <div
                      className="card-image"
                    >
                      <figure
                        className="image is-96x96 has-text-centered"
                      >
                        <svg
                          height="96"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                          />
                        </svg>
                      </figure>
                    </div>
                    <div
                      className="card-content content"
                    >
                      <h3
                        className="has-text-centered"
                      >
                        Elastic
                      </h3>
                      <p>
                        Hive
                        <sup>
                          io
                        </sup>
                         enables auto scaling solutions to save you not only money but increased uptime when production workloads get unpredictable.
                      </p>
                    </div>
                  </article>
                </div>
              </div>
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child"
                >
                  <article
                    className="card m-6 is-reactive"
                  >
                    <div
                      className="card-image"
                    >
                      <figure
                        className="image is-96x96 has-text-centered"
                      >
                        <svg
                          height="96"
                          role="img"
                          viewBox="0 0 512 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288z"
                          />
                        </svg>
                      </figure>
                    </div>
                    <div
                      className="card-content content"
                    >
                      <h3
                        className="has-text-centered"
                      >
                        Message Driven
                      </h3>
                      <p>
                        Hive
                        <sup>
                          io
                        </sup>
                         provides an asynchronous, message passing framework that ensures loose coupling, isolation, and location transparency with Node.js and Kafka.
                      </p>
                    </div>
                  </article>
                </div>
              </div>
            </div>
          </div>
          <div
            className="tile is-parent"
          >
            <div
              className="tile is-child"
            >
              <article
                className="card m-6 is-serverless"
              >
                <div
                  className="card-image"
                >
                  <figure
                    className="image is-96x96 has-text-centered"
                  >
                    <svg
                      height="96"
                      role="img"
                      viewBox="0 0 640 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M537.585 226.56C541.725 215.836 544 204.184 544 192c0-53.019-42.981-96-96-96-19.729 0-38.065 5.954-53.316 16.159C367.042 64.248 315.288 32 256 32c-88.366 0-160 71.634-160 160 0 2.728.07 5.439.204 8.133C40.171 219.845 0 273.227 0 336c0 79.529 64.471 144 144 144h368c70.692 0 128-57.308 128-128 0-61.93-43.983-113.586-102.415-125.44z"
                      />
                    </svg>
                  </figure>
                </div>
                <div
                  className="card-content content"
                >
                  <h3
                    className="has-text-centered"
                  >
                    Cloud Native
                  </h3>
                  <p>
                    Hive
                    <sup>
                      io
                    </sup>
                     is implemented with standardized Docker images to wrap your domain logic with a lightweight, RESTful interface. This allows you to skip the boilerplate setup/maintenance and dive straight into adding immediate value to your application.
                  </p>
                  <p>
                    Each base image is highly extensible and configurable right out of the box. They make very few assumptions to support different architectures which allows it to pair well with other cloud native initiatives and patterns.
                  </p>
                  <p>
                    We've even added embedded observability with the flip of a switch by integrating with OpenTelemetry. This gives immediate visibility into how your services perform with tracing and metrics support.
                  </p>
                </div>
              </article>
            </div>
          </div>
        </div>
        <div
          className="tile is-ancestor"
        >
          <div
            className="tile is-parent is-8"
          >
            <div
              className="tile is-child"
            >
              <article
                className="card m-6 is-robust"
              >
                <div
                  className="card-image"
                >
                  <figure
                    className="image is-96x96 has-text-centered"
                  >
                    <svg
                      height="96"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 168v-16c0-13.255 10.745-24 24-24h360V80c0-21.367 25.899-32.042 40.971-16.971l80 80c9.372 9.373 9.372 24.569 0 33.941l-80 80C409.956 271.982 384 261.456 384 240v-48H24c-13.255 0-24-10.745-24-24zm488 152H128v-48c0-21.314-25.862-32.08-40.971-16.971l-80 80c-9.372 9.373-9.372 24.569 0 33.941l80 80C102.057 463.997 128 453.437 128 432v-48h360c13.255 0 24-10.745 24-24v-16c0-13.255-10.745-24-24-24z"
                      />
                    </svg>
                  </figure>
                </div>
                <div
                  className="card-content content"
                >
                  <h3
                    className="has-text-centered"
                  >
                    Flexible & Robust
                  </h3>
                  <p>
                    Hive
                    <sup>
                      io
                    </sup>
                     uses the JSON Schema and Flux Standard Action specifications to automatically serialize and validate your data Models for network transport. We call it Model JSON Serialization and it allows you to standardize data exchanges directly into fully validated Model instance.
                  </p>
                </div>
              </article>
            </div>
          </div>
          <div
            className="tile is-parent"
          >
            <div
              className="tile is-child has-text-centered-mobile"
            >
              <svg
                className="m-6 is-hive"
                height="72"
                role="img"
                viewBox="0 0 117.31758 135.46667"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m 58.658792,6.7627869 52.802038,30.4852731 0,60.970543 L 58.658789,128.70387 5.8567507,98.2186 l 1.7e-6,-60.970543 z"
                  fill="none"
                  stroke="#daa520"
                  strokeWidth="12"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="hero is-small is-hive"
  >
    <div
      className="hero-body"
    >
      <nav
        className="level is-mobile"
      >
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="next page"
            href="/overview"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </span>
          </a>
        </div>
      </nav>
    </div>
  </section>,
]
`;
